client/me/purchases/purchases-list/purchase-list-concierge-banner.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {
+	CONCIERGE_HAS_AVAILABLE_PURCHASED_SESSION,
+	CONCIERGE_HAS_UPCOMING_APPOINTMENT,
+} from 'calypso/me/concierge/constants';
+import ConciergeBanner from 'calypso/me/purchases/concierge-banner/index';
+
+type Props = {
+	nextAppointment?: {
+		id: number;
+		siteId: number;
+		scheduleId: number;
+	};
+	availableSessions: number[];
+	siteId?: number;
+	isUserBlocked: boolean;
+};
+
+export function PurchaseListConciergeBanner( props: Props ) {
+	const { nextAppointment, availableSessions, siteId, isUserBlocked } = props;
+
+	if ( isUserBlocked ) {
+		return null;
+	}
+
+	const hasAppointment = Boolean( nextAppointment );
+	const hasSiteSelectedAndIsViewingSite = siteId && availableSessions.includes( siteId );
+	const isNotOnSiteButHasSites = ! siteId && availableSessions.length;
+	const shouldShowBanner =
+		hasAppointment || hasSiteSelectedAndIsViewingSite || isNotOnSiteButHasSites;
+
+	if ( ! shouldShowBanner ) {
+		return null;
+	}
+
+	let bannerType;
+
+	if ( nextAppointment ) {
+		bannerType = CONCIERGE_HAS_UPCOMING_APPOINTMENT;
+	} else {
+		bannerType = CONCIERGE_HAS_AVAILABLE_PURCHASED_SESSION;
```

Comment:
Relating to my comment RE collapsing the banner type constants, this usage feels weird because we don't know if the session was purchased or included.

Reply:
I will refactor these constants too. I think `CONCIERGE_HAS_UPCOMING_APPOINTMENT` and `CONCIERGE_HAS_AVAILABLE_SESSION` should be ok - the distinction between purchased or included isn't really important on the front-end.